### PR TITLE
Refactor charm code

### PR DESCRIPTION
**Pull request by arturo-seijas**
Refactor charm code. Also fixes CI on merge

### CODE REVIEW

**Comment by jdkandersson** [comment]
This looks similar to this: https://github.com/canonical/discourse-k8s-operator/pull/40

**Review by yhaliaw — DISMISSED** [review]
LGTM

**Review by weiiwang01 — DISMISSED** [review]

**Review by gtrkiller — APPROVED** [review]
LGTM

File: image/scripts/pod_setup.sh [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
-su -s /bin/bash -c "${CONTAINER_APP_ROOT}/app/bin/bundle exec rake assets:precompile RAILS_ENV=$RAILS_ENV" "${CONTAINER_APP_USERNAME}" 2>&1 | sed 's/^/asset-build: /'
+"${CONTAINER_APP_ROOT}/app/bin/bundle" exec rake --trace db:migrate
+"${CONTAINER_APP_ROOT}/app/bin/bundle" exec rake assets:precompile 2>&1 | sed 's/^/asset-build: /'
 
 if [ -n "${DISCOURSE_USE_S3}" ] && [ "${DISCOURSE_USE_S3}" == "true" ]; then
 	echo "Running migration to S3..."
-	su -s /bin/bash -c "${CONTAINER_APP_ROOT}/app/bin/bundle exec rake s3:upload_assets RAILS_ENV=$RAILS_ENV" "${CONTAINER_APP_USERNAME}" 2>&1
-else
-	echo "Not running migration to S3 (DISCOURSE_USE_S3=${DISCOURSE_USE_S3})."
+	"${CONTAINER_APP_ROOT}/app/bin/bundle" exec rake s3:upload_assets 2>&1
```
Comment (weiiwang01) [inline image/scripts/pod_setup.sh:8]: Should we add prefix to this command output like previous commands?
Something like `| sed 's/^/upload-assets: /'`

**Review by merkata — APPROVED** [review]
Looks good, nice refactoring!

**Pull request merged**